src/data/ordmap/ordset.lean

```diff
@@ -0,0 +1,1432 @@
+/-
+Copyright (c) 2017 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+-/
+import data.ordmap.ordnode
+import algebra.ordered_ring
+import data.nat.dist
+import tactic.linarith
+
+/-!
+# Verification of the `ordnode α` datatype, with a type `ordset α` of verified sets.
+-/
+
+theorem cmp_le_flip {α} [has_le α] [@decidable_rel α (≤)] (x y : α) :
+  @cmp_le (order_dual α) _ _ x y = cmp_le y x := rfl
+
+theorem cmp_le_swap {α} [has_le α] [is_total α (≤)] [@decidable_rel α (≤)] (x y : α) :
+  (cmp_le x y).swap = cmp_le y x :=
+begin
+  simp [cmp_le], split_ifs; try {refl},
+  cases not_or h h_1 (total_of _ _ _)
+end
```

Comment:
Is there any better home for these lemmas? It seems that now `algebra/order` and `order/basic` do not `import` each other but may be we can change this (or just merge these two files).